tensorflow/Ubuntu/tensorflow_1.10.0_gpu_ubuntu_16.04.py

```diff
@@ -0,0 +1,210 @@
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+
+# Build script for TensorFlow 1.10.0 with GPU support on PPC64LE
+# Run from within the docker container: docker.io/nvidia/cuda-ppc64le:9.2-cudnn7-devel-ubuntu16.04
+# Requires nccl_2.2.13-1+cuda9.2_ppc64le.tgz downloaded from https://developer.nvidia.com/nccl/nccl-download
+# be in the same directory as the sript
+
+# To run this script
+# apt-get update -y
+# apt-get install -y python
+# python build_tensorflow_1_10_0_gpu.py
+
+# The resulting wheel file will be in the tensorflow_pkg subdirectory from where the script was invoked.
+
+
+import os.path, sys, subprocess
+from subprocess import check_call, CalledProcessError
+
+
+packages = [
+    'apt-get install -y --no-install-recommends \
+      openjdk-8-jdk \
+      wget \
+      curl \
+      unzip \
+      zip \
+      git \
+      rsync \
+      python-dev \
+      swig \
+      libatlas-dev \
+      python-numpy \
+      libopenblas-dev \
+      libcurl3-dev \
+      libfreetype6-dev \
+      libzmq3-dev \
+      libhdf5-dev \
+      g++ \
+      patch \
+      python-pip \
+      python-setuptools \
+      python-wheel \
+      python-enum34',
+    'pip install mock cython'
+    ]
+# In the Ubuntu 16.04 images, cudnn is placed in system paths. Move them to
+# /usr/local/cuda
+cudnn = [
+    'cp -P /usr/include/cudnn.h /usr/local/cuda/include',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn.so /usr/local/cuda/lib64',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn.so.7 /usr/local/cuda/lib64',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn.so.7.2.1 /usr/local/cuda/lib64',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn_static.a /usr/local/cuda/lib64',
+    'cp -P /usr/lib/powerpc64le-linux-gnu/libcudnn_static_v7.a /usr/local/cuda/lib64',
+    ]
+nccl = [
+    'tar -xzvf nccl_2.2.13-1+cuda9.2_ppc64le.tgz',
+    'cp cuda/targets/ppc64le-linux/include/nccl.h /usr/local/cuda/include',
+    'mkdir -p /usr/local/cuda/lib',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl.so /usr/local/cuda/lib/',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl.so.2 /usr/local/cuda/lib/',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl.so.2.2.13 /usr/local/cuda/lib/',
+    'cp cuda/targets/ppc64le-linux/lib/libnccl_static.a /usr/local/cuda/lib/',
+    'chmod a+r /usr/local/cuda/include/',
+    'chmod a+r /usr/local/cuda/lib/',
+    'ldconfig'
+    ]
+bazel = [
+    'mkdir -p bazel',
+    'wget https://github.com/bazelbuild/bazel/releases/download/0.15.0/bazel-0.15.0-dist.zip',
+    'mv bazel-0.15.0-dist.zip bazel/',
+    'unzip -o bazel/bazel-0.15.0-dist.zip -d bazel/',
+    'bazel/compile.sh',
+    'cp bazel/output/bazel /usr/local/bin/'
+    ]
+git = [
+    'rm -rf tensorflow',
+    'git clone -b v1.10.0 https://github.com/tensorflow/tensorflow',
+    'wget  https://github.com/wdirons/tensorflow/commit/03c24d6f13bb30d87745c6eb752019457863b00c.patch',
```

Comment:
I realized this patch is only necessary for RHEL / gcc 4.8. I'm going to remove it for ubuntu. I'll test that when testing the changes for Jason's comment.